.jenkins/pipelines/code-coverage.Jenkinsfile

```diff
@@ -5,79 +5,325 @@
+        LCOV_FILE =         "lcov-${GIT_COMMIT[0..7]}.info"
+        LCOV_DIR="mystikos-cc-${GIT_COMMIT[0..7]}"
+        BUILD_INFO =      "${BUILD_URL == '' ? 'N/A' : BUILD_URL}"
+        PACKAGE_INSTALL =   "${UBUNTU_VERSION == '20.04' ? 'Ubuntu-2004' : 'Ubuntu-1804'}_${PACKAGE_NAME}.${PACKAGE_EXTENSION}"
+        BUILD_USER = sh(
+            returnStdout: true,
+            script: 'echo \${USER}'
+        )
     }
     stages {
-        stage('Run Code Coverage Tests') {
-            matrix {
-                axes {
-                    axis {
-                        name 'OS_VERSION'
-                        values '18.04'
+        stage("Cleanup files") {
+            steps {
+                sh "${JENKINS_SCRIPTS}/global/clean-temp.sh"
+            }
+        }
+        stage("Checkout Pull Request") {
+            when {
+                expression { params.PULL_REQUEST_ID != "" }
+            }
+            steps {
+                cleanWs()
+                checkout([$class: 'GitSCM',
+                    branches: [[name: "pr/${PULL_REQUEST_ID}"]],
+                    extensions: [],
+                    userRemoteConfigs: [[
+                        url: 'https://github.com/deislabs/mystikos',
+                        refspec: "+refs/pull/${PULL_REQUEST_ID}/merge:refs/remotes/origin/pr/${PULL_REQUEST_ID}"
+                    ]]
+                ])
+            }
+        }
+        stage('Verify commit sync') {
+            when {
+                expression { params.COMMIT_SYNC != "" }
+            }
+            steps {
+                // Check if the checked out commit is the same across all parallel builds
+                script {
+                    def GIT_COMMIT_ID = sh(
+                        returnStdout: true,
+                        script: "git log --max-count=1 --pretty=format:'%H'"
+                    ).trim()
+                    if ( GIT_COMMIT_ID != params.COMMIT_SYNC ) {
+                        error("Checked out commit (${GIT_COMMIT_ID}) does not match commit from upstream job (${params.COMMIT_SYNC})")
+                    } else {
+                        println("Checked out commit (${GIT_COMMIT_ID}) matches upstream job (${params.COMMIT_SYNC}). Continuing with build.")
                     }
-                    axis {
-                        name 'TEST_PIPELINE'
-                        values 'Unit', 'Solutions', 'DotNet', 'Azure-SDK'
+                }
+            }
+        }
+        stage('Init Config') {
+            steps {
+                sh """
+                   # Initialize dependencies repo
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/global/init-config.sh
+
+                   # Install global dependencies
+                   ${JENKINS_SCRIPTS}/global/wait-dpkg.sh
+                   ${JENKINS_SCRIPTS}/global/init-install.sh
+                   """
+            }
+        }
+        stage('Setup AZ CLI') {
```

Comment:
Is az cli used in this pipeline? If not, this stage can be removed.

Reply:
It is used in the azure-sdk tests - the pipeline had failed when this stage was removed